exporter/signalfxexporter/translation/converter.go

```diff
@@ -94,287 +83,273 @@ func NewMetricsConverter(logger *zap.Logger, t *MetricTranslator) *MetricsConver
 
-	return sfxDataPoints, numDroppedTimeSeries
+		val := inDp.Value()
+		dp.Value.IntValue = &val
+
+		out = append(out, &dp)
+	}
+	return out
 }
 
-func appendAttributesToDimensions(
-	dimensions []*sfxpb.Dimension,
-	attribs map[string]string,
-) []*sfxpb.Dimension {
+func convertDoubleDatapoints(in pdata.DoubleDataPointSlice, basePoint *sfxpb.DataPoint, extraDims []*sfxpb.Dimension) []*sfxpb.DataPoint {
+	out := make([]*sfxpb.DataPoint, 0, in.Len())
 
-	for k, v := range attribs {
-		dim := &sfxpb.Dimension{
-			Key:   k,
-			Value: v,
+	for i := 0; i < in.Len(); i++ {
+		inDp := in.At(i)
+		if inDp.IsNil() {
+			continue
 		}
-		dimensions = append(dimensions, dim)
+
+		dp := *basePoint
+		dp.Timestamp = timestampToSignalFx(inDp.Timestamp())
+		dp.Dimensions = labelsToDimensions(inDp.LabelsMap(), extraDims)
+
+		val := inDp.Value()
+		dp.Value.DoubleValue = &val
+
+		out = append(out, &dp)
 	}
-	return dimensions
+	return out
 }
 
-func fromOCMetricDescriptorToMetricType(ocType metricspb.MetricDescriptor_Type) *sfxpb.MetricType {
-	if ocType > metricspb.MetricDescriptor_UNSPECIFIED &&
-		ocType <= metricspb.MetricDescriptor_SUMMARY {
-		return ocDescriptorToMetricType[int(ocType)]
+func makeBaseDataPoint(m pdata.Metric) *sfxpb.DataPoint {
+	return &sfxpb.DataPoint{
+		Metric:     m.Name(),
+		MetricType: fromMetricDataTypeToMetricType(m.DataType()),
 	}
-
-	return nil
 }
 
-func appendDistributionValues(
-	sfxDataPoints []*sfxpb.DataPoint,
-	sfxBaseDataPoint *sfxpb.DataPoint,
-	distributionValue *metricspb.DistributionValue,
-) []*sfxpb.DataPoint {
-
-	// Translating distribution values per symmetrical recommendations to Prometheus:
-	// https://docs.signalfx.com/en/latest/integrations/agent/monitors/prometheus-exporter.html#overview
-
-	// 1. The total count gets converted to a cumulative counter called
-	// <basename>_count.
-	// 2. The total sum gets converted to a cumulative counter called <basename>.
-	sfxDataPoints = appendTotalAndSum(
-		sfxDataPoints,
-		sfxBaseDataPoint,
-		&distributionValue.Count,
-		&distributionValue.Sum)
-
-	// 3. Each histogram bucket is converted to a cumulative counter called
-	// <basename>_bucket and will include a dimension called upper_bound that
-	// specifies the maximum value in that bucket. This metric specifies the
-	// number of events with a value that is less than or equal to the upper
-	// bound.
-	metricName := sfxBaseDataPoint.Metric + "_bucket"
-	explicitBuckets := distributionValue.BucketOptions.GetExplicit()
-	if explicitBuckets == nil {
-		return sfxDataPoints
+func fromMetricDataTypeToMetricType(typ pdata.MetricDataType) *sfxpb.MetricType {
+	switch typ {
+	case pdata.MetricDataTypeIntGauge:
+		return &sfxMetricTypeGauge
+	case pdata.MetricDataTypeDoubleGauge:
+		return &sfxMetricTypeGauge
+	case pdata.MetricDataTypeIntSum:
+		return &sfxMetricTypeCumulativeCounter
+	case pdata.MetricDataTypeDoubleSum:
+		return &sfxMetricTypeCumulativeCounter
```

Comment:
If they are not monotonic in SFX we need to make them Gauges.

Reply:
If they represent a summation of something they should probably still be cumulative counters to get a more natural rollup by default.  The backend already handles sums not being monotonic since you have that issue every time a service restarts and "monotonic" counters reset.

Reply:
That is not correct. If you make them let's say "CumulativeCounters", and I report for one MTS:
[5 7 3]

This is not a restart between 7 and 3, and the signalfx backend will calculate the "Sum" as 10 in the rollup.

Reply:
These are treated as gauges now.